tripy/examples/resent50/resnet50.ipynb

```diff
@@ -0,0 +1,529 @@
+    "|   | | MaxPool 3x3, Stride 2|   | bottlenecks):           |   |               |\n",
+    "|   | | Output: (56 x 56 x 64)|  | - Stage 1: (3 layers)   |   |               |\n",
+    "|   | +----------------------+   | - Stage 2: (4 layers)   |   |               |\n",
+    "|   |                            | - Stage 3: (6 layers)   |   |               |\n",
+    "|   |                            | - Stage 4: (3 layers)   |   |               |\n",
+    "|   |                            | Output: (1 x 1 x 2048)  |   |               |\n",
+    "|   |                            +-------------------------+   |               |\n",
+    "|   |                                            |             |               |\n",
+    "|   |                                            v             |               |\n",
+    "|   |  +----------------------------------------------------+  |               |\n",
+    "|   |  | Average Pooling (kernel 7x7, strid 7x7)            |  |               |\n",
+    "|   |  | Output: (1 x 1 x 2048)                             |  |               |\n",
+    "|   |  +----------------------------------------------------+  |               |\n",
+    "|                               |                                              |\n",
+    "|                               v                                              |\n",
+    "|   +-------------------------------------------------------+                  |\n",
+    "|   |                   Classifier                          |                  |\n",
+    "|   |                 Fully Connected Layer                 |                  | -> Ouput classes probablity (1000)\n",
+    "|   |               Output: (1000 classes)                  |                  |\n",
+    "|   +-------------------------------------------------------+                  |\n",
+    "+-------------------------------------------------------------------------------+\n",
+    "\n",
+    "```"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "class ResNetConvLayer(tp.Module):\n",
+    "    def __init__(self, in_channels, out_channels, kernel_dims, stride=(1, 1), padding=((0, 0), (0, 0)), activation=True):\n",
+    "        super(ResNetConvLayer, self).__init__()\n",
+    "        self.convolution = tp.Conv(\n",
+    "            in_channels, out_channels, kernel_dims=kernel_dims,\n",
+    "            stride=stride, padding=padding, bias=False\n",
+    "        )\n",
+    "        self.normalization = tp.BatchNorm(out_channels)\n",
+    "        self.activation = activation\n",
+    "\n",
+    "    def __call__(self, x):\n",
+    "        x = self.convolution(x)\n",
+    "        x = self.normalization(x)\n",
+    "        if self.activation:\n",
+    "            x = tp.relu(x)\n",
+    "        return x\n",
+    "    \n",
+    "class ResNetShortCut(tp.Module):\n",
```

Comment:
Should each module be its own cell?

Reply:
I think that's not neccessary, we could do one big cell, but that would be a bit ugly. I tried to keep relevant parts together on the same cell.